flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/StandaloneFlinkService.java

```diff
@@ -66,14 +68,14 @@ public StandaloneFlinkService(
     @Override
     protected void deployApplicationCluster(JobSpec jobSpec, Configuration conf) throws Exception {
         LOG.info("Deploying application cluster");
-        submitClusterInternal(conf, Mode.APPLICATION);
+        submitClusterInternal(conf, Mode.APPLICATION, Optional.of(jobSpec));
```

Comment:
Instead of passing the jobSpec through we should stick to the current model of passing the args through conf.

In `FlinkConfigBuilder` we can take the args from the JobSpec and set them to the flink config `kubernetes.jobmanager.entrypoint.args` (already exists).

This can then be accessed through the `StandaloneKubernetesJobManagerParameters` in `CmdStandaloneJobManagerDecorator` to add the args

Reply:
Thanks for the feedback! That makes more sense